hub/api/tensor.py

```diff
@@ -90,8 +95,18 @@ def append(self, array: np.ndarray):
 
         Args:
             array (np.ndarray): The data to add to the tensor.
+            index_map_dict (dict): Dictionary with image bytes and additional image metadata.
```

Comment:
What's going on here? The description doesn't match the parameter name, and this docstring needs some updated examples.

Why did we add this parameter? I feel like it would be cleaner to call append with a PIL image then some dict, much less `append(None, data_dict)`. We can use `isinstance` in that case to have only one argument.

Reply:
yes, please move this one layer of abstraction above